### ISSUE TITLE
devDisabledInDebug not building apk with dev disabled

### ISSUE THREAD

**Issue by Sumit-Chakole** [facebook/react-native#28206]
Opening new issue as https://github.com/facebook/react-native/issues/27090 is closed and issue still exists

```
project.ext.react = [
entryFile: "index.js",
bundleInDebug: true,
devDisabledInDebug: true
]

apply from: "../../node_modules/react-native/react.gradle
```

Apk built using following ways, its still shows development menu on shake

react-native run-android
gradlew assembleDebug
react-native run-android --variant=debug
gradlew bundleDebugJsAndAssets then gradlew assembleDebug

React Native version: 0:61

React Native Environment Info:
System:
OS: Windows 8.1
CPU: (4) x64 Intel(R) Core(TM) i7-5500U CPU @ 2.40GHz
Memory: 1.70 GB / 7.88 GB
Binaries:
npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
IDEs:
Android Studio: Version 3.5.0.0 AI-191.8026.42.35.5900203

## Steps To Reproduce
1. react-native init AwesomeProject
2. Add following in app build.gradle file before react gradle is applied
```
bundleInDebug: true,
devDisabledInDebug: true
```

## Expected Results

Debug build of apk should work without metro bundler and it should not show development menu or yellow warning boxes. This is suppose to be given to our tester

## link to a repository:

[ https://github.com/Sumit-Chakole/dev_disabled_repo2]( https://github.com/Sumit-Chakole/dev_disabled_repo2)

**Comment by Ashoat** [comment]
I am seeing this issue on macOS as well.

**Comment by Sumit-Chakole** [comment]
any update on this?

**Comment by Sumit-Chakole** [comment]
Issue exists, please get it resolved

**Comment by abdullahizzuddiin** [comment]
Is there any update for this issue?

**Comment by abdullahizzuddiin** [comment]
I tried a few command related to my problem from this [article](https://github.com/react-native-community/cli/blob/master/docs/commands.md#bundle). `index.android.bundle` that generated from running `react-native bundle` manually always replaced after I ran `./graddlew assembleDebug`. It was not worked.

But, when I tried this configuration: I added new `buildType` and set `bundleIn...` and `devDisabledIn...` to my new buildType. It work like a charm. My app completely disconnected from development mode.

```
...
project.ext.react = 
[ 
  ... 
  bundleInStaging: true,
  devDisabledInStaging: true,
]
...
android { 
 buildTypes { `
   ...
   staging { 
     signingConfig signingConfigs.debug 
     matchingFallbacks = ['debug'] 
   }
}
...
```

It seems, it is only not worked if I set `bundleIn...` and `devDisabledIn...` to `debug`
